### ISSUE TITLE
Accessibility: Time picker cannot be accessed with keyboard navigation

### ISSUE THREAD

**Issue by greg-a-smith** [Hacker0x01/react-datepicker#1370]
### Expected behavior
Similar to the dates in the calendar, a user should be able to use keyboard navigation to select a time from the TimePicker.

### Actual behavior
The times in the TimePicker (either date and time or time only) cannot be accessed by any combination of keystrokes.

### Steps to reproduce
Go to https://reactdatepicker.com to the **Select Time** and **Select Time Only** examples.

### Suggested direction
In order for the input, the calendar and the time picker to all be properly accessible with keyboard navigation, they need to be decoupled and broken into separate components.  The browser focus actually needs to move from one control to the next so proper key handling can be achieved.  The overlay also likely needs to be linked to a trigger button rather than simply tied to input focus.  When showing an overlay to pick both date and time, it needs to include some form of OK/Cancel buttons since multiple selections can (and need to) be made.

We are interested in continuing to use `react-datepicker`, but we need to make these suggested changes in order to stay compliant.  Our team is willing to help work through these issues (this one plus #1369 and #1371) with multiple pull requests, if you are open to it.

**Comment by jlurena** [comment]
@greg-a-smith  would you be willing to share your fork? I also need an accessible time picker.

**Comment by greg-a-smith** [comment]
@jlurena Our team has actually gone a different direction as our corporate environment shifted to a new set of components so we won't be reliant on react-datepicker anymore.
